codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java

```diff
@@ -132,20 +132,17 @@ public String blobShape(final BlobShape shape) {
       ? "input"
       : dataSource;
     return """
-    func () []byte {
-    b := []byte{}
-    if %s == nil {
-        return nil
-    }
-    for i := dafny.Iterate(%s) ; ; {
-        val, ok := i()
-        if !ok {
-            return b
-        } else {
-            b = append(b, val.(byte))
-        }
+      func () []byte {
+      if %s == nil {
+          return nil
+      }
+      a := input.(dafny.Sequence).ToArray().(dafny.GoNativeArray);
```

Comment:
Should this be `%s.ToArray().(dafny.GoNativeArray)`?

Reply:
In Go, we only need assertion when a variable is interface (`input.(dafny.Sequence)`). When its not interface, we don't need assertion. And there are cases where the dataSource will be a interface and cases where it won't be. So, using datasource is safer.

So, I suspect `input.(dafny.Sequence)` would fail in MPL.

Reply:
FYI -- the wrong version did fine in MPL and ESDK.